oat-sa/tao-item-runner-fe
# Feature/nex 1100/control item runner display

Related tp https://oat-sa.atlassian.net/browse/NEX-1100 
 
  - Added new API to the item runner to control the item display using the following new methods : 
     - `suspend`
     - `close`
     - `resume`
     - `isSuspended`
     - `isClosed`
The semantic is close to the QTI item session lifecycle on purpose. Maybe a next step will be to integrate all the states, but this requires some breaking changes... So let's do it incrementally  

 - The repository has not been updated for a long time, so I did a bit of update and clean up : 
   - update the API code to use ES2015 coding style
   - add shared eslint config
   - add prettier
   - remove lodash from the API
   - reformat the code
   - the `error` event throws real `Error` instead of strings (this is backward compatible, thanks to `Error.toString()`)

To simplify the review, the clean up and changes have been made in separated commits? 

**How to test :** 
 - Run unit test and check coverage
 - update it in TAO's `taoItems` and `taoQtiItem` extensions and try some items in the preview and authoring
